File: 001-020/003.rb
target_num = 600_851_475_143
calc_num = target_num
i = 3
factors = [2]

while i <= calc_num
  unless (target_num % i).zero?
    i += 1
    next
  end
  calc_num /= i
  factors.each do |item|
    unless (i % item).zero?
      factors.push(i)
      break
    end
  end
  i += 1
end

ans = factors[factors.length - 1]

puts ans

File: 001-010/003.py
target_num = 600_851_475_143
calc_num = target_num
i = 3
factors = [2]

while i <= calc_num:
    if target_num % i != 0:
        i += 1
        continue
    calc_num /= i
    for factor in factors:
        if i % factor != 0:
            factors.append(i)
            break
    i += 1

ans = factors[len(factors) - 1]

print(ans)
